internal/brokers/examplebroker/broker.go

```diff
@@ -129,7 +132,7 @@ func (b *Broker) GetAuthenticationModes(ctx context.Context, sessionID string, s
 		allModes = getMfaModes(sessionInfo, sessionInfo.allModes)
 	}
 	// If the user needs or can reset the password, we only show those authentication modes.
-	if sessionInfo.currentAuthStep > 1 && sessionInfo.pwdChange != noReset {
+	if sessionInfo.currentAuthStep > 1 && sessionInfo.currentAuthStep == sessionInfo.neededAuthSteps && sessionInfo.pwdChange != noReset {
```

Comment:
I don’t think (as said in previous comment) that you need to keep currentAuthStep > 1. We present the reset pasword on the last step in that case, done.

Reply:
This is to prevent something like creating a user with `currentAuthStep=1`, `neededAuthSteps=1` and `pwdChange > Reset`. 

Yes, I know we control the creation of users and so on, but this check doesn't impact anything and prevents us from getting diverted to "why am I getting those password reset modes?"

Reply:
On the contrary, let’s say one day you have a password needs reset user and you set the number of MFA steps to 1, you will be equally confused :p

Reply:
So, if you want to really avoid this and be explicit, get the first if as I wrote, and inside,

```
if sessionInfo.currentAuthStep < 2 {
   return an error
}
```

Reply:
That's precisely why we renamed the variable to ``neededAuthSteps`, no? So this confusion that you pointed out never happens. But the `pwdChange` value is not related to the authSteps and can be set independently of it, so we need to have this control flow which associates both.

Reply:
Right, this is why, if you want to ensure that for the user requesting a password change to have at least 2 steps, let’s make that explicit by returning an error.